

<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../file_types/">
      
      
        <link rel="next" href="../../day3/igv_visualisation/">
      
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.11">
    
    
      
        <title>Samtools - NGS - Quality control, Alignment, Visualisation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.4af4bdda.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
<!-- Matomo -->
<script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
      var u="https://matomo.sib.swiss/";
      _paq.push(['setTrackerUrl', u+'matomo.php']);
      _paq.push(['setSiteId', '220']);
      var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
      g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
    })();
  </script>
  <!-- End Matomo Code -->
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  <meta property="og:url"                content="https://sib-swiss.github.io/NGS-introduction-training/latest/" />
  <meta property="og:title"              content="NGS - Quality Control, Alignment, Visualisation" />
  <meta property="og:description"        content="Course on the introduction to next generation sequencing analysis" />
  <meta property="og:image"              content="https://sib-swiss.github.io/NGS-introduction-training/2022.11/assets/images/alignment_infograph.png" />
  
  <meta name="twitter:card" content="summary_large_image" />
  <meta name="twitter:title" content="NGS - Quality Control, Alignment, Visualisation" />
  <meta name="twitter:description" content="Course on the introduction to next generation sequencing analysis" />
  <meta name="twitter:image" content="https://sib-swiss.github.io/NGS-introduction-training/2022.11/assets/images/alignment_infograph.png" />

  
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NGS - Quality control, Alignment, Visualisation" class="md-header__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS - Quality control, Alignment, Visualisation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Samtools
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/NGS-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-introduction-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS - Quality control, Alignment, Visualisation" class="md-nav__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    NGS - Quality control, Alignment, Visualisation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/NGS-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 1
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/intro/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/sequencing_technologies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sequencing technologies
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/server_login/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Setup
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/reproducibility/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reproducibility
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day1/quality_control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quality control
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 2
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../read_alignment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Read alignment
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../file_types/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    File types
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Samtools
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Samtools
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    <span class="md-ellipsis">
      Learning outcomes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    <span class="md-ellipsis">
      Material
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercises" class="md-nav__link">
    <span class="md-ellipsis">
      Exercises
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Exercises">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      Alignment statistics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compression-sorting-and-indexing" class="md-nav__link">
    <span class="md-ellipsis">
      Compression, sorting and indexing
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Filtering
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#redirection" class="md-nav__link">
    <span class="md-ellipsis">
      Redirection
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-summary" class="md-nav__link">
    <span class="md-ellipsis">
      QC summary
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 3
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../day3/igv_visualisation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    IGV and visualisation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../group_work/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Group work
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    <span class="md-ellipsis">
      Learning outcomes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    <span class="md-ellipsis">
      Material
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercises" class="md-nav__link">
    <span class="md-ellipsis">
      Exercises
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Exercises">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      Alignment statistics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compression-sorting-and-indexing" class="md-nav__link">
    <span class="md-ellipsis">
      Compression, sorting and indexing
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Filtering
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#redirection" class="md-nav__link">
    <span class="md-ellipsis">
      Redirection
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-summary" class="md-nav__link">
    <span class="md-ellipsis">
      QC summary
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


  <h1>Samtools</h1>

<h2 id="learning-outcomes">Learning outcomes</h2>
<p><strong>After having completed this chapter you will be able to:</strong></p>
<ul>
<li>Use <code>samtools flagstat</code> to get general statistics on the flags stored in a sam/bam file</li>
<li>Use <code>samtools view</code> to:<ul>
<li>compress a sam file into a bam file</li>
<li>filter on sam flags</li>
<li>count alignments</li>
<li>filter out a region</li>
</ul>
</li>
<li>Use <code>samtools sort</code> to sort an alignment file based on coordinate</li>
<li>Use <code>samtools index</code> to create an index of a sorted sam/bam file</li>
<li>Use the pipe (<code>|</code>) symbol to pipe alignments directly to <code>samtools</code> to perform sorting and filtering</li>
</ul>
<h2 id="material">Material</h2>
<ul>
<li><code>samtools</code> <a href="http://www.htslib.org/doc/samtools.html">documentation</a></li>
<li>Explain sam flags <a href="https://broadinstitute.github.io/picard/explain-flags.html">tool</a></li>
</ul>
<h2 id="exercises">Exercises</h2>
<h3 id="alignment-statistics">Alignment statistics</h3>
<p><strong>Exercise:</strong> Write the statistics of the E. coli alignment to file called <code>SRR519926.sam.stats</code> by using <code>samtools flagstat</code>. Find the documentation <a href="http://www.htslib.org/doc/samtools-flagstat.html">here</a>. Anything that draws your attention?</p>
<details class="done">
<summary>Answer</summary>
<p>Code:
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/project/results/alignments/
samtools<span class="w"> </span>flagstat<span class="w"> </span>SRR519926.sam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.sam.stats
</code></pre></div></p>
<p>resulting in:</p>
<div class="highlight"><pre><span></span><code>624724 + 0 in total (QC-passed reads + QC-failed reads)
624724 + 0 primary
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
621624 + 0 mapped (99.50% : N/A)
621624 + 0 primary mapped (99.50% : N/A)
624724 + 0 paired in sequencing
312362 + 0 read1
312362 + 0 read2
300442 + 0 properly paired (48.09% : N/A)
619200 + 0 with itself and mate mapped
2424 + 0 singletons (0.39% : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ&gt;=5)
</code></pre></div>
<p>Of the reads, 47.87% is properly paired. The rest isn&rsquo;t. Proper pairing is quite hard to interpret. It usually means that the 0x2 flag (each segment properly aligned according to the aligner) is false. In this case it means that the insert size is high for a lot of sequences. That is because the insert size distribution is very wide. You can find info on insert size distribution like this:</p>
<div class="highlight"><pre><span></span><code>samtools stats SRR519926.sam | grep ^SN | cut -f 2,3
</code></pre></div>
<p>Now look at <code>insert size average</code> and <code>insert size standard deviation</code>. You can see the standard deviation is higher than the average, suggesting a wide distribution.</p>
</details>
<h3 id="compression-sorting-and-indexing">Compression, sorting and indexing</h3>
<p>The command <code>samtools view</code> is very versatile. It takes an alignment file and writes a filtered or processed alignment to the output. You can for example use it to compress your SAM file into a BAM file. Let&rsquo;s start with that.</p>
<p><strong>Exercise</strong>: Create a script called <code>08_compress_sort.sh</code>. Add a <code>samtools view</code> command to compress our SAM file into a BAM file and include the header in the output. For this, use the <code>-b</code> and <code>-h</code> options. Find the required documentation <a href="http://www.htslib.org/doc/samtools-view.html">here</a>. How much was the disk space reduced by compressing the file?</p>
<div class="admonition tip">
<p class="admonition-title">Tip: Samtools writes to stdout</p>
<p>By default, samtools writes it&rsquo;s output to stdout. This means that you need to redirect your output to a file with <code>&gt;</code> or use the the output option <code>-o</code>.</p>
</div>
<details class="done">
<summary>Answer</summary>
<p><div class="highlight"><span class="filename">08_compress_sort.sh</span><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>

<span class="nb">cd</span><span class="w"> </span>~/project/results/alignments

samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>SRR519926.sam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.bam
</code></pre></div>
By using <code>ls -lh</code>, you can find out that <code>SRR519926.sam</code> has a size of 264 Mb, while <code>SRR519926.bam</code> is only 77 Mb.  </p>
</details>
<p>To look up specific alignments, it is convenient to have your alignment file indexed. An indexing can be compared to a kind of &lsquo;phonebook&rsquo; of your sequence alignment file. Indexing is done with <code>samtools</code> as well, but it first needs to be sorted on coordinate (i.e. the alignment location). You can do it like this:</p>
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>sort<span class="w"> </span>SRR519926.bam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.sorted.bam
samtools<span class="w"> </span>index<span class="w"> </span>SRR519926.sorted.bam
</code></pre></div>
<p><strong>Exercise</strong>: Add these lines to <code>08_compress_sort.sh</code>, and re-run te script in order to generate the sorted bam file. After that checkout the headers of the unsorted bam file (<code>SRR519926.bam</code>) and the sorted bam file (<code>SRR519926.sorted.bam</code>) with <code>samtools view -H</code>. What are the differences?</p>
<details class="done">
<summary>Answer</summary>
<p>Your script should like like this:</p>
<div class="highlight"><span class="filename">08_compress_sort.sh</span><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>

<span class="nb">cd</span><span class="w"> </span>~/project/results/alignments

samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>SRR519926.sam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.bam
samtools<span class="w"> </span>sort<span class="w"> </span>SRR519926.bam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.sorted.bam
samtools<span class="w"> </span>index<span class="w"> </span>SRR519926.sorted.bam
</code></pre></div>
<p><code>samtools view -H SRR519926.bam</code> returns:</p>
<div class="highlight"><pre><span></span><code>@HD     VN:1.0  SO:unsorted
@SQ     SN:U00096.3     LN:4641652
@PG     ID:bowtie2      PN:bowtie2      VN:2.4.2        CL:&quot;/opt/conda/envs/ngs-tools/bin/bowtie2-align-s --wrapper basic-0 -x /config/project/ref_genome//ecoli-strK12-MG1655.fasta -1 /config/project/trimmed_data/trimmed_SRR519926_1.fastq -2 /config/project/trimmed_data/trimmed_SRR519926_2.fastq&quot;
@PG     ID:samtools     PN:samtools     PP:bowtie2      VN:1.12 CL:samtools view -bh SRR519926.sam
@PG     ID:samtools.1   PN:samtools     PP:samtools     VN:1.12 CL:samtools view -H SRR519926.bam
</code></pre></div>
<p>And <code>samtools view -H SRR519926.sorted.bam</code> returns:</p>
<div class="highlight"><pre><span></span><code>@HD     VN:1.0  SO:coordinate
@SQ     SN:U00096.3     LN:4641652
@PG     ID:bowtie2      PN:bowtie2      VN:2.4.2        CL:&quot;/opt/conda/envs/ngs-tools/bin/bowtie2-align-s --wrapper basic-0 -x /config/project/ref_genome//ecoli-strK12-MG1655.fasta -1 /config/project/trimmed_data/trimmed_SRR519926_1.fastq -2 /config/project/trimmed_data/trimmed_SRR519926_2.fastq&quot;
@PG     ID:samtools     PN:samtools     PP:bowtie2      VN:1.12 CL:samtools view -bh SRR519926.sam
@PG     ID:samtools.1   PN:samtools     PP:samtools     VN:1.12 CL:samtools sort SRR519926.bam
@PG     ID:samtools.2   PN:samtools     PP:samtools.1   VN:1.12 CL:samtools view -H SRR519926.sorted.bam
</code></pre></div>
<p>There are two main differences: </p>
<ul>
<li>The <code>SO</code> tag at <code>@HD</code> type code has changed from <code>unsorted</code> to <code>coordinate</code>.</li>
<li>A line with the <code>@PG</code> type code for the sorting was added.</li>
</ul>
<p>Note that the command to view the header (<code>samtools -H</code>) is also added to the header for both runs.</p>
</details>
<h3 id="filtering">Filtering</h3>
<p>With <code>samtools view</code> you can easily filter your alignment file based on flags. One thing that might be sensible to do at some point is to filter out unmapped reads.</p>
<p><strong>Exercise:</strong> Check out the flag that you would need to filter for mapped reads. It&rsquo;s at page 7 of the <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM documentation</a>.</p>
<details class="done">
<summary>Answer</summary>
<p>You will need the 0x4 flag.</p>
</details>
<p>Filtering against unmapped reads (leaving only mapped reads) with <code>samtools view</code> would look like this:</p>
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>-F<span class="w"> </span>0x4<span class="w"> </span>SRR519926.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.sorted.mapped.bam
</code></pre></div>
<p>or:</p>
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>-F<span class="w"> </span><span class="m">4</span><span class="w"> </span>SRR519926.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.sorted.mapped.bam
</code></pre></div>
<p><strong>Exercise:</strong> Generate a script called <code>09_extract_unmapped.sh</code> to get only the unmapped reads (so the opposite of the example). How many reads are in there? Is that the same as what we expect based on the output of <code>samtools flagstat</code>?</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Check out the <code>-f</code> and <code>-c</code> options of <code>samtools view</code></p>
</div>
<details class="done">
<summary>Answer</summary>
<p>Your script <code>09_extract_unmapped.sh</code> should look like this:</p>
<div class="highlight"><span class="filename">09_extract_unmapped.sh</span><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>

<span class="nb">cd</span><span class="w"> </span>~/project/results/alignments

samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>-f<span class="w"> </span>0x4<span class="w"> </span>SRR519926.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>SRR519926.sorted.unmapped.bam
</code></pre></div>
<p>Counting like this:
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>view<span class="w"> </span>-c<span class="w"> </span>SRR519926.sorted.unmapped.bam
</code></pre></div></p>
<p>This should correspond to the output of <code>samtools flagstat</code> (624724 - 621624 = 3100)</p>
</details>
<p><code>samtools view</code> also enables you to filter alignments in a specific region. This can be convenient if you don&rsquo;t want to work with huge alignment files and if you&rsquo;re only interested in alignments in a particular region. Region filtering only works for sorted and indexed alignment files.</p>
<p><strong>Exercise:</strong> Generate a script called <code>10_extract_region.sh</code> to filter our sorted and indexed BAM file for the region between 2000 and 2500 kb, and output it as a BAM file with a header.</p>
<div class="admonition tip">
<p class="admonition-title">Tip: Specifying a region</p>
<p>Our E. coli genome has only one chromosome, because only one line starts with <code>&gt;</code> in the fasta file</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>~/project/ref_genome
grep<span class="w"> </span><span class="s2">&quot;&gt;&quot;</span><span class="w"> </span>ecoli-strK12-MG1655.fasta
</code></pre></div>
<p>gives:</p>
<div class="highlight"><pre><span></span><code>&gt;U00096.3 Escherichia coli str. K-12 substr. MG1655, complete genome
</code></pre></div>
<p>The part after the first space in the title is cut off for the alignment reference. So the code for specifying a region would be: <code>U00096.3:START-END</code></p>
</div>
<details class="done">
<summary>Answer</summary>
<div class="highlight"><span class="filename">10_extract_region.sh</span><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>

<span class="nb">cd</span><span class="w"> </span>~/project/results/alignments

samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span><span class="se">\</span>
SRR519926.sorted.bam<span class="w"> </span><span class="se">\</span>
U00096.3:2000000-2500000<span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span>SRR519926.sorted.region.bam
</code></pre></div>
</details>
<h3 id="redirection">Redirection</h3>
<p>Samtools is easy to use in a pipe. In this case you can replace the input file with a <code>-</code>. For example, you can sort and compress the output of your alignment software in a pipe like this:</p>
<div class="highlight"><pre><span></span><code>my_alignment_command<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>-<span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span>alignment.bam
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">The use of <code>-</code></p>
<p>In the modern versions of samtools, the use of <code>-</code> is not needed for most cases, so without an input file it reads from stdin. However, if you&rsquo;re not sure, it&rsquo;s better to be safe than sorry.</p>
</div>
<p><strong>Exercise:</strong> Write a script called <code>11_align_sort.sh</code> that maps the reads with bowtie2 (see chapter 2 of <a href="../read_alignment/">read alignment</a>), sorts them, and outputs them as a BAM file with a header.</p>
<details class="done">
<summary>Answer</summary>
<div class="highlight"><span class="filename">11_align_sort.sh</span><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>

<span class="nv">TRIMMED_DIR</span><span class="o">=</span>~/project/results/trimmed
<span class="nv">REFERENCE_DIR</span><span class="o">=</span>~/project/ref_genome
<span class="nv">ALIGNED_DIR</span><span class="o">=</span>~/project/results/alignments

bowtie2<span class="w"> </span><span class="se">\</span>
-x<span class="w"> </span><span class="nv">$REFERENCE_DIR</span>/ecoli-strK12-MG1655.fasta<span class="w"> </span><span class="se">\</span>
-1<span class="w"> </span><span class="nv">$TRIMMED_DIR</span>/trimmed_SRR519926_1.fastq<span class="w"> </span><span class="se">\</span>
-2<span class="w"> </span><span class="nv">$TRIMMED_DIR</span>/trimmed_SRR519926_2.fastq<span class="w"> </span><span class="se">\</span>
<span class="m">2</span>&gt;<span class="w"> </span><span class="nv">$ALIGNED_DIR</span>/bowtie2_SRR519926.log<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>-<span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="nv">$ALIGNED_DIR</span>/SRR519926.sorted.mapped.frompipe.bam
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Redirecting <code>stderr</code></p>
<p>Notice the line starting with <code>2&gt;</code>. This redirects standard error to a file: <code>$ALIGNED_DIR/bowtie2_SRR519926.log</code>. This file now contains the bowtie2 logs, that can later be re-read or used in e.g. <code>multiqc</code>. </p>
</div>
</details>
<h3 id="qc-summary">QC summary</h3>
<p>The software <a href="https://multiqc.info/">MultiQC</a> is great for creating summaries out of log files and reports from many different bioinformatic tools (including <code>fastqc</code>, <code>fastp</code>, <code>samtools</code> and <code>bowtie2</code>). You can specify a directory that contains any log files, and it will automatically search it for you. </p>
<p><strong>Exercise</strong>: Run the command <code>multiqc .</code> in <code>~/project</code> and checkout the generated report. </p>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.f8cc74c7.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.c8b220af.min.js"></script>
      
    
  </body>
</html>