
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="shortcut icon" href="../../assets/images/dna-svgrepo-com.svg">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.0.2">
    
    
      
        <title>Group work - organisation - NGS - Quality control, Alignment, Visualisation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.38780c08.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.3f72e892.min.css">
        
      
    
    
    
      
        
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
      
  
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#roles-organisation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="../.." title="NGS - Quality control, Alignment, Visualisation" class="md-header-nav__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M.1 494.1c-1.1 9.5 6.3 17.8 15.9 17.8l32.3.1c8.1 0 14.9-5.9 16-13.9.7-4.9 1.8-11.1 3.4-18.1H380c1.6 6.9 2.9 13.2 3.5 18.1 1.1 8 7.9 14 16 13.9l32.3-.1c9.6 0 17.1-8.3 15.9-17.8-4.6-37.9-25.6-129-118.9-207.7-17.6 12.4-37.1 24.2-58.5 35.4 6.2 4.6 11.4 9.4 17 14.2H159.7c21.3-18.1 47-35.6 78.7-51.4C410.5 199.1 442.1 65.8 447.9 17.9 449 8.4 441.6.1 432 .1L399.6 0c-8.1 0-14.9 5.9-16 13.9-.7 4.9-1.8 11.1-3.4 18.1H67.8c-1.6-7-2.7-13.1-3.4-18.1-1.1-8-7.9-14-16-13.9L16.1.1C6.5.1-1 8.4.1 17.9 5.3 60.8 31.4 171.8 160 256 31.5 340.2 5.3 451.2.1 494.1zM224 219.6c-25.1-13.7-46.4-28.4-64.3-43.6h128.5c-17.8 15.2-39.1 30-64.2 43.6zM355.1 96c-5.8 10.4-12.8 21.1-21 32H114c-8.3-10.9-15.3-21.6-21-32h262.1zM92.9 416c5.8-10.4 12.8-21.1 21-32h219.4c8.3 10.9 15.4 21.6 21.2 32H92.9z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            NGS - Quality control, Alignment, Visualisation
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Group work - organisation
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header-nav__source">
        
<a href="https://github.com/sib-swiss/NGS-introduction-training" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-introduction-training
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS - Quality control, Alignment, Visualisation" class="md-nav__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M.1 494.1c-1.1 9.5 6.3 17.8 15.9 17.8l32.3.1c8.1 0 14.9-5.9 16-13.9.7-4.9 1.8-11.1 3.4-18.1H380c1.6 6.9 2.9 13.2 3.5 18.1 1.1 8 7.9 14 16 13.9l32.3-.1c9.6 0 17.1-8.3 15.9-17.8-4.6-37.9-25.6-129-118.9-207.7-17.6 12.4-37.1 24.2-58.5 35.4 6.2 4.6 11.4 9.4 17 14.2H159.7c21.3-18.1 47-35.6 78.7-51.4C410.5 199.1 442.1 65.8 447.9 17.9 449 8.4 441.6.1 432 .1L399.6 0c-8.1 0-14.9 5.9-16 13.9-.7 4.9-1.8 11.1-3.4 18.1H67.8c-1.6-7-2.7-13.1-3.4-18.1-1.1-8-7.9-14-16-13.9L16.1.1C6.5.1-1 8.4.1 17.9 5.3 60.8 31.4 171.8 160 256 31.5 340.2 5.3 451.2.1 494.1zM224 219.6c-25.1-13.7-46.4-28.4-64.3-43.6h128.5c-17.8 15.2-39.1 30-64.2 43.6zM355.1 96c-5.8 10.4-12.8 21.1-21 32H114c-8.3-10.9-15.3-21.6-21-32h262.1zM92.9 416c5.8-10.4 12.8-21.1 21-32h219.4c8.3 10.9 15.4 21.6 21.2 32H92.9z"/></svg>

    </a>
    NGS - Quality control, Alignment, Visualisation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/sib-swiss/NGS-introduction-training" title="Go to repository" class="md-source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 00-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 01-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 000 40.81l195.61 195.6a28.86 28.86 0 0040.8 0l194.69-194.69a28.86 28.86 0 000-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../precourse/" class="md-nav__link">
      Precourse preparations
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../course_schedule/" class="md-nav__link">
      Course schedule
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Day 1
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Day 1" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        <span class="md-nav__icon md-icon"></span>
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/intro/" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/sequencing_technologies/" class="md-nav__link">
      Sequencing technologies
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/server_login/" class="md-nav__link">
      Server login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/quality_control/" class="md-nav__link">
      Quality control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/read_alignment/" class="md-nav__link">
      Read alignment
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Day 2
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Day 2" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        <span class="md-nav__icon md-icon"></span>
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../file_types/" class="md-nav__link">
      File types
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../samtools/" class="md-nav__link">
      Samtools
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../igv_visualisation/" class="md-nav__link">
      IGV and visualisation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Group work - organisation
        <span class="md-nav__icon md-icon"></span>
      </label>
    
    <a href="./" class="md-nav__link md-nav__link--active">
      Group work - organisation
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#roles-organisation" class="md-nav__link">
    Roles &amp; organisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-1-short-read-rna-seq-of-mice" class="md-nav__link">
     Project 1: Short-read RNA-seq of mice.
  </a>
  
    <nav class="md-nav" aria-label=" Project 1: Short-read RNA-seq of mice.">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks" class="md-nav__link">
    Tasks:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" class="md-nav__link">
    Questions:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-2-short-read-rna-seq-data-of-arabidopsis-thaliana-grown-in-space" class="md-nav__link">
     Project 2: Short-read RNA-seq data of Arabidopsis thaliana grown in space
  </a>
  
    <nav class="md-nav" aria-label=" Project 2: Short-read RNA-seq data of Arabidopsis thaliana grown in space">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_1" class="md-nav__link">
    Tasks:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" class="md-nav__link">
    Questions:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-3-long-read-genome-sequencing" class="md-nav__link">
     Project 3: Long-read genome sequencing
  </a>
  
    <nav class="md-nav" aria-label=" Project 3: Long-read genome sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_2" class="md-nav__link">
    Tasks:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" class="md-nav__link">
    Questions:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../day3/" class="md-nav__link">
      Day 3
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#roles-organisation" class="md-nav__link">
    Roles &amp; organisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-1-short-read-rna-seq-of-mice" class="md-nav__link">
     Project 1: Short-read RNA-seq of mice.
  </a>
  
    <nav class="md-nav" aria-label=" Project 1: Short-read RNA-seq of mice.">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks" class="md-nav__link">
    Tasks:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" class="md-nav__link">
    Questions:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-2-short-read-rna-seq-data-of-arabidopsis-thaliana-grown-in-space" class="md-nav__link">
     Project 2: Short-read RNA-seq data of Arabidopsis thaliana grown in space
  </a>
  
    <nav class="md-nav" aria-label=" Project 2: Short-read RNA-seq data of Arabidopsis thaliana grown in space">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_1" class="md-nav__link">
    Tasks:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" class="md-nav__link">
    Questions:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-3-long-read-genome-sequencing" class="md-nav__link">
     Project 3: Long-read genome sequencing
  </a>
  
    <nav class="md-nav" aria-label=" Project 3: Long-read genome sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_2" class="md-nav__link">
    Tasks:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" class="md-nav__link">
    Questions:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Group work - organisation</h1>
                
                <p>The last part of this course will consist of project-based-learning. This means that you will work in groups on a single question. We will split up into groups of five people.</p>
<div class="admonition note">
<p class="admonition-title">If online</p>
<p>If the course takes place online, we will use break-out rooms to communicate within groups. Please stay in the break-out room during the day, also if you are working individually.</p>
</div>
<h2 id="roles-organisation">Roles &amp; organisation</h2>
<p>Project based learning is about learning by doing, but also about <em>peer instruction</em>. This means that you will be both a learner and a teacher. There will be differences in levels among participants, but because of that, some will learn efficiently from people that have just learned, and others will teach and increase their understanding.</p>
<p>Each project has <strong>tasks</strong> and <strong>questions</strong>. By performing the tasks, you should be able to answer the questions. At the start of the project, make sure that each of you gets a task assigned. You should consider the tasks and questions as a guidance. If interesting questions pop up during the project, you are <strong>encouraged</strong> to work on those. Also, you don&rsquo;t have to perform all the tasks and answer all the questions.</p>
<p>In the afternoon of day 2, you will divide the initial tasks, and start on the project. On day 3, you can work on the project in the morning and in the first part of the afternoon. We will conclude the projects with a <strong>10-minute presentation</strong> of each group.</p>
<h2 id="project-1-short-read-rna-seq-of-mice"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M472.29 195.9l-67.06-23c-19.28-6.6-33.54-20.92-38.14-38.31l-16-60.45c-11.58-43.77-76.57-57.13-110-22.62L195 99.24c-13.26 13.71-33.54 20.93-54.2 19.31l-71.9-5.62c-52-4.07-86.93 44.89-59 82.84l38.54 52.42c11.08 15.07 12.82 33.86 4.64 50.24l-28.43 57C4 396.67 47.46 440.29 98.11 429.23l70-15.28c20.11-4.39 41.45 0 57.07 11.73l54.32 40.83c39.32 29.56 101 7.57 104.45-37.22l4.7-61.86c1.35-17.8 12.8-33.87 30.63-43l62-31.74c44.84-22.96 39.55-80.17-8.99-96.79zM160 256a32 32 0 1132-32 32 32 0 01-32 32zm128 96a32 32 0 1132-32 32 32 0 01-32 32zm16-128a16 16 0 1116-16 16 16 0 01-16 16z"/></svg></span> Project 1: Short-read RNA-seq of mice.</h2>
<p><strong>Aim:</strong> Compare <code>hisat2</code> (splice-aware) with <code>bwa mem</code> (splice unaware) while aligning a mouse RNA-seq dataset.</p>
<p>In this project you will be working with data from:</p>
<p>Singhania A, Graham CM, Gabryšová L, Moreira-Teixeira L, Stavropoulos E, Pitt JM, et al (2019). <em>Transcriptional profiling unveils type I and II interferon networks in blood and tissues across diseases</em>. Nat Commun. 10:1–21. <a href="https://doi.org/10.1038/s41467-019-10601-6">https://doi.org/10.1038/s41467-019-10601-6</a></p>
<p>Here&rsquo;s the <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA490485">BioProject page</a>.</p>
<h3 id="tasks">Tasks:</h3>
<ul>
<li>Check out the BioProject, and download two samples that interest you.</li>
<li>Do a QC on the data with <code>fastqc</code></li>
<li>Check which options to use, and run <code>bwa-mem</code></li>
<li>Check which options to use, and run <code>hisat2</code></li>
<li>Find out how to measure computational speed, and measure it</li>
<li>Evaluate the alignment quality (e.g. alignment rates, mapping quality)</li>
<li>Compare the bam files of the two aligners in IGV</li>
<li>Compare different samples in read quality, alignment rates, depth, etc.</li>
<li>Run <code>featurecounts</code> on both alignments</li>
<li>Compare the count matrices in <code>R</code> or <code>python</code></li>
</ul>
<h3 id="questions">Questions:</h3>
<ul>
<li>Is there a difference in computational speed?</li>
<li>What are the alignment rates?</li>
<li>How do the aligners handle splicing?</li>
<li>How are spliced alignments stored in the SAM file?</li>
<li>Do you see differences in soft clipping?</li>
<li>What would be the effect of the aligner if you would be measuring gene expression? (To investigate this you&rsquo;ll need to run e.g. <a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a>).</li>
</ul>
<div class="admonition hint">
<p class="admonition-title">Downloading from SRA</p>
<div class="highlight"><pre><span></span><code>prefetch <span class="o">[</span>SRR number<span class="o">]</span>
fastq-dump --split-files --gzip <span class="o">[</span>SRR number<span class="o">]</span>
</code></pre></div>
</div>
<div class="admonition hint">
<p class="admonition-title">Example code hisat2</p>
<p>Everything in between <code>&lt;&gt;</code> should be replaced with specific arguments</p>
<div class="highlight"><pre><span></span><code>hisat2-build &lt;reference_sequence_fasta&gt; &lt;index_basename&gt;

hisat2 <span class="se">\</span>
-x &lt;index_basename&gt; <span class="se">\</span>
-1 &lt;mate1.fastq.gz&gt; <span class="se">\</span>
-2 &lt;mate2.fastq.gz&gt; <span class="se">\</span>
-p &lt;threads&gt; <span class="se">\</span>
<span class="p">|</span> samtools sort <span class="se">\</span>
<span class="p">|</span> samtools view -bh <span class="se">\</span>
&gt; &lt;alignment_file.bam&gt;
</code></pre></div>
</div>
<div class="admonition hint">
<p class="admonition-title">Spliced alignments</p>
<p>Have a look at IGV on a particular gene, e.g. Nbr1</p>
</div>
<h2 id="project-2-short-read-rna-seq-data-of-arabidopsis-thaliana-grown-in-space"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M64 96H0c0 123.7 100.3 224 224 224v144c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V320C288 196.3 187.7 96 64 96zm384-64c-84.2 0-157.4 46.5-195.7 115.2 27.7 30.2 48.2 66.9 59 107.6C424 243.1 512 147.9 512 32h-64z"/></svg></span> Project 2: Short-read RNA-seq data of <em>Arabidopsis thaliana</em> grown in space</h2>
<p><strong>Aim:</strong> Compare <code>hisat2</code> (splice-aware) with <code>bowtie2</code> (splice unaware) while aligning an Arabidopsis RNA-seq dataset.</p>
<p>The analysis of this dataset is reported in this paper:</p>
<p>Vandenbrink JP, Herranz R, Poehlman WL, Alex Feltus F, Villacampa A, Ciska M, et al. (2019) <em>RNA-seq analyses of Arabidopsis thaliana seedlings after exposure to blue-light phototropic stimuli in microgravity </em>. Am J Bot. 106:1466–76.</p>
<p>Reads can be found on the NASA repository <a href="https://genelab-data.ndc.nasa.gov/genelab/accession/GLDS-251/">here</a>. You can download data with <code>wget</code>. Here is an example for downloading the forward and reverse reads for sample 131:</p>
<div class="highlight"><pre><span></span><code>wget -O sample_131_R1.fastq.gz <span class="se">\</span>
https://genelab-data.ndc.nasa.gov/genelab/static/media/dataset/GLDS-251_rna-seq_13JUN2017HiSeq_Run_Sample_131_UMISS_Hoeksema_ACAGTG_L001_R1_001.fastq.gz?version<span class="o">=</span><span class="m">1</span>

wget -O sample_131_R2.fastq.gz <span class="se">\</span>
https://genelab-data.ndc.nasa.gov/genelab/static/media/dataset/GLDS-251_rna-seq_13JUN2017HiSeq_Run_Sample_131_UMISS_Hoeksema_ACAGTG_L001_R2_001.fastq.gz?version<span class="o">=</span><span class="m">1</span>
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Use of <code>-O</code></p>
<p>The name of the file is very long we use the option <code>-O</code> here to give it a shorter name.</p>
</div>
<h3 id="tasks_1">Tasks:</h3>
<ul>
<li>Check out the BioProject, and download two samples that interest you.</li>
<li>Do a QC on the data with <code>fastqc</code></li>
<li>Check which options to use, and run <code>bwa-mem</code></li>
<li>Check which options to use, and run <code>hisat2</code></li>
<li>Find out how to measure computational speed, and measure it</li>
<li>Evaluate the alignment quality (e.g. alignment rates, mapping quality)</li>
<li>Compare the bam files of the two aligners in IGV</li>
<li>Compare different samples in read quality, alignment rates, depth, etc.</li>
<li>Run <code>featurecounts</code> on both alignments</li>
<li>Compare the count matrices in <code>R</code> or <code>python</code></li>
</ul>
<h3 id="questions_1">Questions:</h3>
<ul>
<li>Is there a difference in computational speed?</li>
<li>What are the alignment rates?</li>
<li>How do the aligners handle splicing?</li>
<li>How are spliced alignments stored in the SAM file?</li>
<li>Do you see differences in soft clipping?</li>
<li>What would be the effect of the aligner if you would be measuring gene expression? (To investigate this you&rsquo;ll need to run e.g. <a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a>).</li>
</ul>
<div class="admonition hint">
<p class="admonition-title">Example code hisat2</p>
<p>Everything in between <code>&lt;&gt;</code> should be replaced with specific arguments</p>
<div class="highlight"><pre><span></span><code>hisat2-build &lt;reference_sequence_fasta&gt; &lt;index_basename&gt;

hisat2 <span class="se">\</span>
-x &lt;index_basename&gt; <span class="se">\</span>
-1 &lt;mate1.fastq.gz&gt; <span class="se">\</span>
-2 &lt;mate2.fastq.gz&gt; <span class="se">\</span>
-p &lt;threads&gt; <span class="se">\</span>
<span class="p">|</span> samtools sort <span class="se">\</span>
<span class="p">|</span> samtools view -bh <span class="se">\</span>
&gt; &lt;alignment_file.bam&gt;
</code></pre></div>
</div>
<h2 id="project-3-long-read-genome-sequencing"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><path d="M208 0c-29.9 0-54.7 20.5-61.8 48.2-.8 0-1.4-.2-2.2-.2-35.3 0-64 28.7-64 64 0 4.8.6 9.5 1.7 14C52.5 138 32 166.6 32 200c0 12.6 3.2 24.3 8.3 34.9C16.3 248.7 0 274.3 0 304c0 33.3 20.4 61.9 49.4 73.9-.9 4.6-1.4 9.3-1.4 14.1 0 39.8 32.2 72 72 72 4.1 0 8.1-.5 12-1.2 9.6 28.5 36.2 49.2 68 49.2 39.8 0 72-32.2 72-72V64c0-35.3-28.7-64-64-64zm368 304c0-29.7-16.3-55.3-40.3-69.1 5.2-10.6 8.3-22.3 8.3-34.9 0-33.4-20.5-62-49.7-74 1-4.5 1.7-9.2 1.7-14 0-35.3-28.7-64-64-64-.8 0-1.5.2-2.2.2C422.7 20.5 397.9 0 368 0c-35.3 0-64 28.6-64 64v376c0 39.8 32.2 72 72 72 31.8 0 58.4-20.7 68-49.2 3.9.7 7.9 1.2 12 1.2 39.8 0 72-32.2 72-72 0-4.8-.5-9.5-1.4-14.1 29-12 49.4-40.6 49.4-73.9z"/></svg></span> Project 3: Long-read genome sequencing</h2>
<p><strong>Aim</strong>: Align long reads from RNA-seq data to a reference genome.</p>
<p>In this project, you will be working with data from:</p>
<p>Clark, M. B. et al (2020). <em>Long-read sequencing reveals the complex splicing profile of the psychiatric risk gene CACNA1C in human brain</em>. Molecular Psychiatry, 25(1), 37–47. <a href="https://doi.org/10.1038/s41380-019-0583-1">https://doi.org/10.1038/s41380-019-0583-1</a>.</p>
<p>Here you can find the <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB34660">BioProject page</a>.</p>
<p>It is Oxford Nanopore Technology sequencing data of amplicons of the gene CACNA1C. It is primarily used to discover new splice variants. In this project, we will align a few of the samples to the reference genome, and assess the quality of reads and the alignment.</p>
<h3 id="tasks_2">Tasks:</h3>
<ul>
<li>Check out the BioProject, and download two samples that interest you.</li>
<li>Perform QC with <code>fastqc</code></li>
<li>Perform QC with <code>NanoPlot</code></li>
<li>Align with <code>minimap2</code> with default parameters</li>
<li>Figure how you should set parameters <code>-x</code> and <code>-G</code></li>
<li>Evaluate the alignment quality (e.g. alignment rates, mapping quality)</li>
<li>Compare different samples in read quality, alignment rates, depth, etc.</li>
<li><strong>Bonus</strong> see if you can identify different splice variants using <a href="https://github.com/BrooksLabUCSC/flair">FLAIR</a> (see hints below).</li>
</ul>
<h3 id="questions_2">Questions:</h3>
<ul>
<li>Have a look at the quality report. What are the average read lengths? Is that expected?</li>
<li>What is the average read quality? What kind of accuracy would you expect?</li>
<li>Note any differences between <code>fastqc</code> and <code>NanoPlot</code>? How is that compared to the publication?</li>
<li>Check out the options <code>-x</code> and <code>-G</code> of <code>minimap2</code>. Are the defaults appropriate?</li>
<li>You might consider using <code>-x map-ont</code> or <code>-x splice</code>. Do you see differences in the alignment in e.g. IGV?</li>
<li>How are spliced alignments stored in the SAM file with the different settings of <code>-x</code>?</li>
<li>How deep is the gene sequenced?</li>
</ul>
<div class="admonition hint">
<p class="admonition-title">Downloading from SRA</p>
<div class="highlight"><pre><span></span><code>prefetch <span class="o">[</span>SRR number<span class="o">]</span>
fastq-dump --gzip <span class="o">[</span>SRR number<span class="o">]</span>
</code></pre></div>
</div>
<div class="admonition hint">
<p class="admonition-title">Intron sizes</p>
<p>Check out the the intron sizes of CACNA1C in e.g. IGV or UCSC genome browser. How does that relate to the parameter <code>-G</code>?</p>
</div>
<div class="admonition hint">
<p class="admonition-title">Accuracy from quality scores</p>
<p>Find the equation to calculate error probability from quality score on <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Wikipedia</a>.</p>
</div>
<div class="admonition hint">
<p class="admonition-title">Comparing <code>fastqc</code> and <code>Nanoplot</code></p>
<p>For comparing <code>fastqc</code> and <code>NanoPlot</code>, check out this <a href="https://gigabaseorgigabyte.wordpress.com/2017/06/26/averaging-basecall-quality-scores-the-right-way/">blog</a> of the author of NanoPlot, and this <a href="https://github.com/wdecoster/NanoPlot/issues/191">thread</a>.</p>
</div>
<div class="admonition hint">
<p class="admonition-title">Running FLAIR</p>
<p>FLAIR is a set of python scripts that can be used to identify and quantify (new) isoforms based on alignment files of long-read sequencing data. You can basically follow the pipeline as described <a href="https://github.com/BrooksLabUCSC/flair">here</a>.</p>
<p>To use FLAIR first clone the git repository, and activate the (pre-configured) conda environment:</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span>
git clone https://github.com/BrooksLabUCSC/flair.git
conda activate flair_env
</code></pre></div>
<p>After that, generate a FLAIR working directory and convert your <code>.bam</code> alignment file to <code>bed12</code> format.</p>
<div class="highlight"><pre><span></span><code>mkdir ~/flair_output

python3 ~/flair/bin/bam2Bed12.py <span class="se">\</span>
-i alignment.bam <span class="se">\</span>
&gt; alignment.bed12
</code></pre></div>
<p>Generate (and run) a shell script to run the modules <code>flair.py correct</code>, <code>flair.py collapse</code> and <code>flair.py quantify</code>. To do this, carefully follow the manual at https://github.com/BrooksLabUCSC/flair. Structure and document your script(s), so you can easily re-run the analysis.</p>
<p>Files you will need are:</p>
<ul>
<li>Reference genome (chromosome 12 only): <code>/data/references/GRCh38.p13.chr12.fa</code></li>
<li>GTF: <code>/data/references/Homo_sapiens.GRCh38.100.gtf</code></li>
<li>Reads manifest: <code>/data/reads/lrrnaseq/batch_combined/reads_manifest.tsv</code></li>
</ul>
</div>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../igv_visualisation/" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                IGV and visualisation
              </div>
            </div>
          </a>
        
        
          <a href="../../day3/" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Day 3
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/vendor.77e55a48.min.js"></script>
      <script src="../../assets/javascripts/bundle.9554a270.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: "../..",
          features: [],
          search: Object.assign({
            worker: "../../assets/javascripts/worker/search.4ac00218.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>