

<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../day3/igv_visualisation/">
      
      
      
      <link rel="icon" href="../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.11">
    
    
      
        <title>Group work - NGS - Quality control, Alignment, Visualisation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.4af4bdda.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
<!-- Matomo -->
<script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
      var u="https://matomo.sib.swiss/";
      _paq.push(['setTrackerUrl', u+'matomo.php']);
      _paq.push(['setSiteId', '220']);
      var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
      g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
    })();
  </script>
  <!-- End Matomo Code -->
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  <meta property="og:url"                content="https://sib-swiss.github.io/NGS-introduction-training/latest/" />
  <meta property="og:title"              content="NGS - Quality Control, Alignment, Visualisation" />
  <meta property="og:description"        content="Course on the introduction to next generation sequencing analysis" />
  <meta property="og:image"              content="https://sib-swiss.github.io/NGS-introduction-training/2022.11/assets/images/alignment_infograph.png" />
  
  <meta name="twitter:card" content="summary_large_image" />
  <meta name="twitter:title" content="NGS - Quality Control, Alignment, Visualisation" />
  <meta name="twitter:description" content="Course on the introduction to next generation sequencing analysis" />
  <meta name="twitter:image" content="https://sib-swiss.github.io/NGS-introduction-training/2022.11/assets/images/alignment_infograph.png" />

  
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#material" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NGS - Quality control, Alignment, Visualisation" class="md-header__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS - Quality control, Alignment, Visualisation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Group work
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/NGS-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-introduction-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NGS - Quality control, Alignment, Visualisation" class="md-nav__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation" data-md-component="logo">
      
  <img src="../assets/images/SIB_logo.svg" alt="logo">

    </a>
    NGS - Quality control, Alignment, Visualisation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/NGS-introduction-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-introduction-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 1
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day1/intro/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day1/sequencing_technologies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sequencing technologies
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day1/server_login/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Setup
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day1/reproducibility/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reproducibility
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day1/quality_control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quality control
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 2
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day 2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day2/read_alignment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Read alignment
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day2/file_types/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    File types
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day2/samtools/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Samtools
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 3
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Day 3
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../day3/igv_visualisation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    IGV and visualisation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Group work
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Group work
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    <span class="md-ellipsis">
      Material
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#roles-organisation" class="md-nav__link">
    <span class="md-ellipsis">
      Roles &amp; organisation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-directories" class="md-nav__link">
    <span class="md-ellipsis">
      Working directories
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-1-variant-analysis-of-human-data" class="md-nav__link">
    <span class="md-ellipsis">
      Project 1: Variant analysis of human data
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Project 1: Variant analysis of human data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" class="md-nav__link">
    <span class="md-ellipsis">
      Questions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hints" class="md-nav__link">
    <span class="md-ellipsis">
      Hints
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-2-long-read-genome-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      Project 2: Long-read genome sequencing
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Project 2: Long-read genome sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_1" class="md-nav__link">
    <span class="md-ellipsis">
      Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" class="md-nav__link">
    <span class="md-ellipsis">
      Questions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-3-short-read-rna-seq-of-mice" class="md-nav__link">
    <span class="md-ellipsis">
      Project 3: Short-read RNA-seq of mice.
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Project 3: Short-read RNA-seq of mice.">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_2" class="md-nav__link">
    <span class="md-ellipsis">
      Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" class="md-nav__link">
    <span class="md-ellipsis">
      Questions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hints_1" class="md-nav__link">
    <span class="md-ellipsis">
      Hints
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    <span class="md-ellipsis">
      Material
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#roles-organisation" class="md-nav__link">
    <span class="md-ellipsis">
      Roles &amp; organisation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-directories" class="md-nav__link">
    <span class="md-ellipsis">
      Working directories
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-1-variant-analysis-of-human-data" class="md-nav__link">
    <span class="md-ellipsis">
      Project 1: Variant analysis of human data
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Project 1: Variant analysis of human data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks" class="md-nav__link">
    <span class="md-ellipsis">
      Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" class="md-nav__link">
    <span class="md-ellipsis">
      Questions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hints" class="md-nav__link">
    <span class="md-ellipsis">
      Hints
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-2-long-read-genome-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      Project 2: Long-read genome sequencing
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Project 2: Long-read genome sequencing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_1" class="md-nav__link">
    <span class="md-ellipsis">
      Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" class="md-nav__link">
    <span class="md-ellipsis">
      Questions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-3-short-read-rna-seq-of-mice" class="md-nav__link">
    <span class="md-ellipsis">
      Project 3: Short-read RNA-seq of mice.
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Project 3: Short-read RNA-seq of mice.">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tasks_2" class="md-nav__link">
    <span class="md-ellipsis">
      Tasks
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" class="md-nav__link">
    <span class="md-ellipsis">
      Questions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hints_1" class="md-nav__link">
    <span class="md-ellipsis">
      Hints
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


  <h1>Group work</h1>

<p>The last part of this course will consist of project-based-learning. This means that you will work in groups on a single question. We will split up into groups of five people.</p>
<div class="admonition note">
<p class="admonition-title">If working with Docker</p>
<p>If you are working with Docker, I assume you are working independently and therefore can not work in a group. However, you can test your skills with these real biological datasets. Realize that the datasets and calculations are (much) bigger compared to the exercises, so check if your computer is up for it. You&rsquo;ll probably need around 4 cores, 16G of RAM and 50G of harddisk.</p>
</div>
<div class="admonition note">
<p class="admonition-title">If online</p>
<p>If the course takes place online, we will use break-out rooms to communicate within groups. Please stay in the break-out room during the day, also if you are working individually.</p>
</div>
<h2 id="material">Material</h2>
<p><a class="md-button" href="../assets/pdf/07_group_work.pdf"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0h160v128c0 17.7 14.3 32 32 32h128v144H176c-35.3 0-64 28.7-64 64v144H64c-35.3 0-64-28.7-64-64zm384 64H256V0zM176 352h32c30.9 0 56 25.1 56 56s-25.1 56-56 56h-16v32c0 8.8-7.2 16-16 16s-16-7.2-16-16V368c0-8.8 7.2-16 16-16m32 80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-16v48zm96-80h32c26.5 0 48 21.5 48 48v64c0 26.5-21.5 48-48 48h-32c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16m32 128c8.8 0 16-7.2 16-16v-64c0-8.8-7.2-16-16-16h-16v96zm80-112c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v32h32c8.8 0 16 7.2 16 16s-7.2 16-16 16h-32v48c0 8.8-7.2 16-16 16s-16-7.2-16-16V368"/></svg></span> Download the presentation</a></p>
<h2 id="roles-organisation">Roles &amp; organisation</h2>
<p>Project based learning is about learning by doing, but also about <em>peer instruction</em>. This means that you will be both a learner and a teacher. There will be differences in levels among participants, but because of that, some will learn efficiently from people that have just learned, and others will teach and increase their understanding.</p>
<p>Each project has <strong>tasks</strong> and <strong>questions</strong>. By performing the tasks, you should be able to answer the questions. You should consider the tasks and questions as a guidance. If interesting questions pop up during the project, you are <strong>encouraged</strong> to work on those. Also, you don&rsquo;t have to perform all the tasks and answer all the questions.</p>
<p>In the afternoon of day 1, you will start on the project. On day 3, you can work on the project in the morning and in the first part of the afternoon. We will conclude the projects with a <strong>10-minute presentation</strong> of each group.</p>
<h2 id="working-directories">Working directories</h2>
<p>Each group has access to a shared working directory. It is mounted in the root directory (<code>/</code>). You can add this directory to your working space by clicking: <strong>File &gt; Add Folder to Workspace&hellip;</strong>. Then, type the path to your group directory: <code>/group_work/groupX</code> (where <code>X</code> is your group number).</p>
<h2 id="project-1-variant-analysis-of-human-data">Project 1: Variant analysis of human data</h2>
<p><strong>Aim</strong>: Find variants on chromosome 20 from three samples</p>
<p>In this project you will be working with Illumina reads from three samples: a father, mother and a child. You will perform quality control, align the reads, mark duplicates, detect variants and visualize them. </p>
<p>You can get the data by running these commands:</p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://ngs-introduction-training.s3.eu-central-1.amazonaws.com/project1.tar.gz
tar<span class="w"> </span>-xvf<span class="w"> </span>project1.tar.gz
rm<span class="w"> </span>project1.tar.gz
</code></pre></div>
<h3 id="tasks">Tasks</h3>
<div class="admonition warning">
<p class="admonition-title">Important!</p>
<p><strong>Stick to the principles for reproducible analysis</strong> described <a href="../day1/reproducibility/">here</a></p>
</div>
<ul>
<li>Download the required data</li>
<li>Do a QC on the data with <code>fastqc</code></li>
<li>Trim adapters and low quality bases with <code>fastp</code>. Make sure to include the option <code>--detect_adapter_for_pe</code>. To prevent overwriting <code>fastp.html</code>, specify a report filename for each sample with the option <code>--html</code>. </li>
<li>After trimming the adapters, run <code>fastqc</code> again to see whether all adapters are gone.</li>
<li>Create an index for bowtie2. At the same time create a fasta index (<code>.fai</code> file) with <code>samtools faidx</code>. </li>
<li>Check which options to use, and align with <code>bowtie2</code>. At the same time add readgroups to the aligned reads (see hints below). Make sure you end up with an indexed and sorted bam file. </li>
<li>Mark duplicates on the individual bam files with <code>gatk MarkDuplicates</code> (see hints below).</li>
<li>Merge the three bam files with <code>samtools merge</code>. Index the bam file afterwards. </li>
<li>Run <code>freebayes</code> to call variants. Only call variants on the region <code>chr20:10018000-10220000</code> by specifying the <code>-r</code> option. </li>
<li>Load your alignments together with the vcf containing the variants in IGV. Check out e.g. <code>chr20:10,026,397-10,026,638</code>. </li>
<li>Run <code>multiqc</code> to get an overall quality report.</li>
</ul>
<h3 id="questions">Questions</h3>
<ul>
<li>Have a look at the quality of the reads. Are there any adapters in there? Did adapter trimming change that? How is the base quality? Could you improve that?</li>
<li>How many duplicates were in the different samples (hint: use <code>samtools flagstat</code>)? Why is it important to remove them for variant analysis?</li>
<li>Why did you add read groups to the bam files? Where is this information added in the bam file? </li>
<li>Are there variants that look spurious? What could be the cause of that? What information in the vcf can you use to evaluate variant quality? </li>
<li>There are two high quality variants in <code>chr20:10,026,397-10,026,638</code>. What are the genotypes of the three samples according to freebayes? Is this according to what you see in the alignments? If the alternative alleles are present in the same individual, are they in phase or in repulsion? Note: you can also load vcf files in IGV. </li>
</ul>
<h3 id="hints">Hints</h3>
<p>You can add readgroups to the alignment file with <code>bowtie2</code> with the options <code>--rg-id</code> and <code>--rg</code>, e.g. (<code>$SAMPLE</code> is a variable containing a sample identifier):</p>
<div class="highlight"><pre><span></span><code>bowtie2<span class="w"> </span><span class="se">\</span>
-x<span class="w"> </span>ref.fa<span class="w"> </span><span class="se">\</span>
-1<span class="w"> </span>r1.fastq.gz<span class="w"> </span><span class="se">\</span>
-2<span class="w"> </span>r2.fastq.gz<span class="w"> </span><span class="se">\</span>
--rg-id<span class="w"> </span><span class="nv">$SAMPLE</span><span class="w"> </span><span class="se">\</span>
--rg<span class="w"> </span>SM:<span class="nv">$SAMPLE</span><span class="w"> </span><span class="se">\</span>
</code></pre></div>
<p>To run <code>gatk MarkDuplicates</code> you will only need to specify <code>--INPUT</code> and <code>--OUTPUT</code>, e.g.:</p>
<div class="highlight"><pre><span></span><code>gatk<span class="w"> </span>MarkDuplicates<span class="w"> </span><span class="se">\</span>
--INPUT<span class="w"> </span>sample.bam<span class="w"> </span><span class="se">\</span>
--OUTPUT<span class="w"> </span>sample.md.bam<span class="w"> </span><span class="se">\</span>
--METRICS_FILE<span class="w"> </span>sample.metrics.txt<span class="w"> </span>
</code></pre></div>
<h2 id="project-2-long-read-genome-sequencing"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M184 0c30.9 0 56 25.1 56 56v400c0 30.9-25.1 56-56 56-28.9 0-52.7-21.9-55.7-50.1-5.2 1.4-10.7 2.1-16.3 2.1-35.3 0-64-28.7-64-64 0-7.4 1.3-14.6 3.6-21.2C21.4 367.4 0 338.2 0 304c0-31.9 18.7-59.5 45.8-72.3C37.1 220.8 32 207 32 192c0-30.7 21.6-56.3 50.4-62.6C80.8 123.9 80 118 80 112c0-29.9 20.6-55.1 48.3-62.1 3-28 26.8-49.9 55.7-49.9m144 0c28.9 0 52.6 21.9 55.7 49.9C411.5 56.9 432 82 432 112c0 6-.8 11.9-2.4 17.4 28.8 6.2 50.4 31.9 50.4 62.6 0 15-5.1 28.8-13.8 39.7 27.1 12.8 45.8 40.4 45.8 72.3 0 34.2-21.4 63.4-51.6 74.8 2.3 6.6 3.6 13.8 3.6 21.2 0 35.3-28.7 64-64 64-5.6 0-11.1-.7-16.3-2.1-3 28.2-26.8 50.1-55.7 50.1-30.9 0-56-25.1-56-56V56c0-30.9 25.1-56 56-56"/></svg></span> Project 2: Long-read genome sequencing</h2>
<p><strong>Aim</strong>: Align long reads from RNA-seq data to a reference genome.</p>
<p>In this project, you will be working with data from:</p>
<blockquote>
<p>Padilla, Juan-Carlos A., Seda Barutcu, Ludovic Malet, Gabrielle Deschamps-Francoeur, Virginie Calderon, Eunjeong Kwon, and Eric Lécuyer. “Profiling the Polyadenylated Transcriptome of Extracellular Vesicles with Long-Read Nanopore Sequencing.” BMC Genomics 24, no. 1 (September 22, 2023): 564. https://doi.org/10.1186/s12864-023-09552-6.</p>
</blockquote>
<p>The authors used RNA sequencing with Oxford Nanopore Technology of both extracellular vesicles and whole cells from cell culture. For this project, we will work with two samples of this study, <code>EV_2</code> (extracellular vesicle) and <code>Cell_2</code> (whole cell). Download and unpack the data files.</p>
<p>Download the human reference genome like this:</p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://ngs-introduction-training.s3.eu-central-1.amazonaws.com/project2.tar.gz
tar<span class="w"> </span>-xvf<span class="w"> </span>project2.tar.gz
rm<span class="w"> </span>project2.tar.gz
</code></pre></div>
<p>You can find the fastq files in the <code>reads</code> folder and the reference genome and its annotation in the <code>reference</code> folder. To reduce computational times we work with a subset of the data on a subset of the genome (chromosome 5 and X).</p>
<h3 id="tasks_1">Tasks</h3>
<div class="admonition warning">
<p class="admonition-title">Important!</p>
<p><strong>Stick to the principles for reproducible analysis</strong> described <a href="../day1/reproducibility/">here</a></p>
</div>
<ul>
<li>Perform QC with <code>fastqc</code></li>
<li>Perform QC with <code>NanoPlot</code></li>
<li>Align with <code>minimap2</code> with default parameters</li>
<li>Figure how you should set parameter <code>-x</code></li>
<li>Evaluate the alignment quality (e.g. alignment rates, mapping quality)</li>
<li>Compare the two different samples in read quality, alignment rates, depth, etc.</li>
<li>Check out the alignments in IGV. Check out e.g. <code>ELOVL5</code>.</li>
</ul>
<h3 id="questions_1">Questions</h3>
<ul>
<li>Have a look at the quality report. What are the average read lengths? Is that expected?</li>
<li>What is the average read quality? What kind of accuracy would you expect?</li>
<li>Note any differences between <code>fastqc</code> and <code>NanoPlot</code>? How is that compared to the publication?</li>
<li>Check out the option <code>-x</code> of <code>minimap2</code>. Are the defaults appropriate?</li>
<li>You might consider using <code>-x map-ont</code> or <code>-x splice</code>. Do you see differences in the alignment in e.g. IGV?</li>
<li>How are spliced alignments stored in the SAM file with the different settings of <code>-x</code>?</li>
<li>How deep is the gene <code>ELOVL5</code> sequenced in both samples?</li>
<li>Do you already see evidence for splice variants in the alignments?</li>
</ul>
<div class="admonition hint">
<p class="admonition-title">Accuracy from quality scores</p>
<p>Find the equation to calculate error probability from quality score on <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Wikipedia</a>.</p>
</div>
<div class="admonition hint">
<p class="admonition-title">Comparing <code>fastqc</code> and <code>Nanoplot</code></p>
<p>For comparing <code>fastqc</code> and <code>NanoPlot</code>, check out this <a href="https://gigabaseorgigabyte.wordpress.com/2017/06/26/averaging-basecall-quality-scores-the-right-way/">blog</a> of the author of NanoPlot, and this <a href="https://github.com/wdecoster/NanoPlot/issues/191">thread</a>.</p>
</div>
<div class="admonition hint">
<p class="admonition-title">Running <code>minimap2</code></p>
<p>Here&rsquo;s an example command for <code>minimap2</code>:</p>
<div class="highlight"><pre><span></span><code>minimap2<span class="w"> </span><span class="se">\</span>
-a<span class="w"> </span><span class="se">\</span>
-x<span class="w"> </span><span class="o">[</span>PARAMETER<span class="o">]</span><span class="w"> </span><span class="se">\</span>
<span class="o">[</span>REFERENCE<span class="o">]</span>.fa<span class="w"> </span><span class="se">\</span>
<span class="o">[</span>FASTQFILE<span class="o">]</span>.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span>&gt;<span class="w"> </span><span class="o">[</span>OUTPUT<span class="o">]</span>.bam
</code></pre></div>
</div>
<h2 id="project-3-short-read-rna-seq-of-mice"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M236.4 61.4 227 75.5c-21.3 32-59.4 48.5-97.3 42.1l-59.6-9.9C33.4 101.6 0 129.9.1 167.1c0 15.9 6.4 31.2 17.6 42.5l29.2 29.2c11 11 17.2 25.9 17.2 41.5 0 15.8-6.4 30.9-17.7 42l-13.1 12.8C22.2 345.9 16 360.7 16 376.2c0 36.8 34.1 64.2 70.1 56.2l62.3-13.8c7.7-1.7 15.7-2.6 23.6-2.6h10c27.2 0 53.7 9.3 75 26.3l30.8 24.7c10.5 8.4 23.6 13 37 13 32.7 0 59.3-26.5 59.3-59.3v-25.2c0-34.9 21.4-66.2 53.9-78.8l36.9-14.3c22.4-8.7 37.2-30.3 37.2-54.3 0-28.1-20.1-52.3-47.8-57.3l-28-5.1c-36.5-6.7-65.4-34.5-73.6-70.7l-7.1-31.5C348.9 53.4 322.1 32 291.3 32c-22 0-42.6 11-54.9 29.4M160 192a32 32 0 1 1 0 64 32 32 0 1 1 0-64m128 16a16 16 0 1 1 32 0 16 16 0 1 1-32 0m0 80a32 32 0 1 1 0 64 32 32 0 1 1 0-64"/></svg></span> Project 3: Short-read RNA-seq of mice.</h2>
<p><strong>Aim:</strong> Generate a count matrix to estimate differential gene expression. </p>
<p>In this project you will be working with data from:</p>
<p>Singhania A, Graham CM, Gabryšová L, Moreira-Teixeira L, Stavropoulos E, Pitt JM, et al (2019). <em>Transcriptional profiling unveils type I and II interferon networks in blood and tissues across diseases</em>. Nat Commun. 10:1–21. <a href="https://doi.org/10.1038/s41467-019-10601-6">https://doi.org/10.1038/s41467-019-10601-6</a></p>
<p>Here&rsquo;s the <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA490485">BioProject page</a>. Since the mouse genome is rather large, we have prepared reads for you that originate from chromosome 5. Use those for the project. Download them like this:</p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://ngs-introduction-training.s3.eu-central-1.amazonaws.com/project3.tar.gz
tar<span class="w"> </span>-xvf<span class="w"> </span>project3.tar.gz
rm<span class="w"> </span>project3.tar.gz
</code></pre></div>
<h3 id="tasks_2">Tasks</h3>
<div class="admonition warning">
<p class="admonition-title">Important!</p>
<p><strong>Stick to the principles for reproducible analysis</strong> described <a href="../day1/reproducibility/">here</a></p>
</div>
<ul>
<li>Download the tar file, and find out what&rsquo;s in the data folder</li>
<li>Do a QC on the fastq files with <code>fastqc</code></li>
<li>Trim adapters and low quality bases with <code>fastp</code></li>
<li>After trimming the adapters, run <code>fastqc</code> again to see whether all adapters are gone.</li>
<li>Check which options to use, and align with <code>hisat2</code></li>
<li>Evaluate the alignment quality (e.g. alignment rates, mapping quality)</li>
<li>Have a look at the alignments in IGV, e.g. check out <code>Sparcl1</code>. For this, you can use the built-in genome (<em>Mouse (mm10)</em>). Do you see any evidence for differential splicing?</li>
<li>Run <code>featureCounts</code> on both alignments. Have a look at the option <code>-Q</code>. For further suggestions, see the hints below. </li>
<li>Compare the count matrices in <code>R</code> (find a script to get started <a href="https://github.com/sib-swiss/NGS-introduction-training/blob/main/scripts/project3/07_run_DESeq2.R">here</a>; Rstudio server is running on the same machine. Approach it with your credentials and username <code>rstudio</code>)</li>
</ul>
<h3 id="questions_2">Questions</h3>
<ul>
<li>Check the description at the SRA sample page. What kind of sample is this?</li>
<li>How does the quality of the reads look? Anything special about the overrepresented sequences? (Hint: <a href="https://blast.ncbi.nlm.nih.gov/">blast</a> some overrepresented sequences, and see what they are)</li>
<li>Did trimming improve the QC results? What could be the cause of the warnings/errors in the <code>fastqc</code> reports?</li>
<li>What are the alignment rates?</li>
<li>How are spliced alignments stored in the SAM file?</li>
<li>Are there any differences between the treatments in the percentage of assigned alignments by <code>featureCounts</code>? What is the cause of this? </li>
<li>Can you find any genes that seem to be differentially expressed? </li>
<li>What is the effect of setting the option <code>-Q</code> in <code>featureCounts</code>?</li>
</ul>
<h3 id="hints_1">Hints</h3>
<p>We are now doing computations on a full genome, with full transcriptomic data. This is quite a bit more than we have used during the exercises. Therefore, computations take longer. However, most tools support parallel processing, in which you can specify how many cores you want to use to run in parallel. Your environment contains <strong>four</strong> cores, so this is also the maximum number of processes you can specify. Below you can find the options used in each command to specify multi-core processing.</p>
<table>
<thead>
<tr>
<th>command</th>
<th>option</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>bowtie2-build</code></td>
<td><code>--threads</code></td>
</tr>
<tr>
<td><code>hisat2-build</code></td>
<td><code>--threads</code></td>
</tr>
<tr>
<td><code>fastqc</code></td>
<td><code>--threads</code></td>
</tr>
<tr>
<td><code>cutadapt</code></td>
<td><code>--cores</code></td>
</tr>
<tr>
<td><code>bowtie2</code></td>
<td><code>--threads</code></td>
</tr>
<tr>
<td><code>hisat2</code></td>
<td><code>--threads</code></td>
</tr>
<tr>
<td><code>featureCounts</code></td>
<td><code>-T</code></td>
</tr>
</tbody>
</table>
<p>Here&rsquo;s some example code for <code>hisat2</code> and <code>featureCounts</code>. Everything in between <code>&lt;&gt;</code> should be replaced with specific arguments.</p>
<p>Here&rsquo;s an example for <code>hisat2</code>:</p>
<div class="highlight"><pre><span></span><code>hisat2-build<span class="w"> </span>&lt;reference_sequence_fasta&gt;<span class="w"> </span>&lt;index_basename&gt;

hisat2<span class="w"> </span><span class="se">\</span>
-x<span class="w"> </span>&lt;index_basename&gt;<span class="w"> </span><span class="se">\</span>
-1<span class="w"> </span>&lt;foward_reads.fastq.gz&gt;<span class="w"> </span><span class="se">\</span>
-2<span class="w"> </span>&lt;reverse_reads.fastq.gz&gt;<span class="w"> </span><span class="se">\</span>
-p<span class="w"> </span>&lt;threads&gt;<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>view<span class="w"> </span>-bh<span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span>&lt;alignment_file.bam&gt;
</code></pre></div>
<p>Example code <code>featureCounts</code>:</p>
<div class="highlight"><pre><span></span><code>featureCounts<span class="w"> </span><span class="se">\</span>
-p<span class="w"> </span><span class="se">\</span>
-T<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="se">\</span>
-a<span class="w"> </span>&lt;annotations.gtf&gt;<span class="w"> </span><span class="se">\</span>
-o<span class="w"> </span>&lt;output.counts.txt&gt;<span class="w"> </span><span class="se">\</span>
*.bam
</code></pre></div>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.f8cc74c7.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.c8b220af.min.js"></script>
      
    
  </body>
</html>